31 extra setup

bishax opened the pull request:
Closes #31

bishax committed:
Repo structural additions

bishax committed:
Preinstall script to unencrypt repo before installing

bishax committed:
Remove tox.ini

bishax commented on research_daps/requirements.txt:
```diff
@@ -0,0 +1,8 @@
+mysql-connector-python==8.0.22
+daps_utils @ git+https://github.com/nestauk/daps_utils@dev
```
Should this be `dev`?

jaklinger replied:
You mean the bit before the @? That's just a label I think

bishax replied:
Sorry this was a note to myself while I was sorting some stuff out.

I was pointing out to myself that I shouldn't pin against `dev` because breaking changes could sneak in.

bishax commented on research_daps/requirements.txt:
```diff
@@ -0,0 +1,8 @@
+mysql-connector-python==8.0.22
+daps_utils @ git+https://github.com/nestauk/daps_utils@dev
+toolz
+numpy
+pandas
+pymysql
+tqdm
```
VERSIONS!

bishax committed:
Pin versions, drop pandas

bishax committed:
Update version requirement. 3.7 is okay

bishax committed:
Fixing pip install

- git-crypt lock in setup.py
- post-install hook for pip editable
- Include (encryped) config/ for sdist/wheel

Also formatting.

bishax committed:
Pin daps_utils to actual commit

bishax committed:
postinstall hook on regular install too

bishax committed:
Include VERSION in sdist

bishax committed:
linting (config + fixes) + swap to poetry (inc. git-crypt adaptations)

bishax committed:
Format initplus

jaklinger commented:
@bishax is this good to close? Just cleaning up the ol' inbox before I leave and I can see that this is dangling

bishax committed:
Merge branch 'dev' into 31_extra_setup
